packages/product-json-to-xlsx/src/writer.js

```diff
@@ -0,0 +1,144 @@
+/* @flow */
+import fs from 'fs'
+import tmp from 'tmp'
+import path from 'path'
+import slugify from 'slugify'
+import archiver from 'archiver'
+import EmitOnce from 'single-emit'
+import { mapHeaders } from '@commercetools/product-json-to-csv'
+import mapValues from './map-values'
+import ProductExcel from './product-excel'
+
+// Accept a highland stream and write the output to a single file
+export function writeToSingleXlsxFile(
+  productStream: Object,
+  output: stream$Writable,
+  logger: Object,
+  headers: Array<Object>
+) {
+  const mappedHeaders = mapHeaders(headers)
+  const headerNames = mappedHeaders.map(header => header.label)
+
+  const excel = new ProductExcel(output)
+  excel.writeHeader(headerNames)
+
+  productStream
+    .each(product => excel.writeRow(mapValues(mappedHeaders, product)))
+    .done(() => {
+      // finish workbook and close the stream
+      excel.finish()
+      logger.info('All products have been written to XLSX file')
+    })
+}
+
+export function onStreamsFinished(
+  streams: Array<stream$Writable>,
+  cb: (err: ?Object) => void
+) {
+  const emitOnce = new EmitOnce(streams, 'finish')
+  emitOnce.on('error', err => cb(err))
+  // call callback when all streams are finished
+  emitOnce.on('finish', () => cb())
+}
+
+export function archiveDir(
+  dir: string,
+  output: stream$Writable,
+  logger: Object
+) {
+  const archive = archiver('zip')
+  archive.on('error', err => {
+    logger.error(err)
+    output.emit('error', err)
+  })
+  archive.pipe(output)
+  archive.directory(dir, 'products')
+  archive.finalize()
+}
+
+export async function finishWorksheetsAndArchive(
```

Comment:
Nitpick: `async` can be removed since we you don't use `await` inside the function

Reply:
updated here f10d31d